File: csharp/1663.smallest-string-with-a-given-numeric-value.cs
/*
 * @lc app=leetcode id=1663 lang=csharp
 *
 * [1663] Smallest String With A Given Numeric Value
 */

using System;

// @lc code=start
public class Solution {
    public string GetSmallestString(int n, int k) {
        char[] ret = new char[n];
        k -= n;
        for (int i = 0; i < n; ++i) {
            ret[i] = (char)((int)'a' + Math.Min(25, Math.Max(0, k - 25 * (n - 1 - i))));
        }
        return new string(ret);
    }
}
// @lc code=end

File: java/1663.smallest-string-with-a-given-numeric-value.java
/*
 * @lc app=leetcode id=1663 lang=java
 *
 * [1663] Smallest String With A Given Numeric Value
 */

// @lc code=start
class Solution {
    public String getSmallestString(int n, int k) {
        if (k < n || k > 26 * n) {
            return "";
        }
        StringBuilder sb = new StringBuilder(n);
        while (n-- > 0) {
            char c;
            if (n * 26 >= k) {
                c = 'a';
            } else if ((n + 1) * 26 == k) {
                c = 'z';
            } else {
                c = (char)('a' + k - 1 - 26 * n);
            }
            sb.append(c);
            k -= c - 'a' + 1;
        }
        return sb.toString();
    }
}
// @lc code=end

